listenbrainz/webserver/static/js/src/listens/ListenCard.tsx

```diff
@@ -14,25 +14,78 @@ import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
 import { getArtistLink, getTrackLink } from "../utils";
 import Card from "../components/Card";
 import APIService from "../APIService";
+import ListenControl from "./ListenControl";
 
 export const DEFAULT_COVER_ART_URL = "/static/img/default_cover_art.png";
 
 export type ListenCardProps = {
+  user: {
+    id?: string;
+    name: string;
+    auth_token: string;
+  };
+  apiUrl: string;
   listen: Listen;
   mode: ListensListMode;
   className?: string;
   playListen: (listen: Listen) => void;
 };
 
-export default class ListenCard extends React.Component<ListenCardProps> {
+type ListenCardState = {
+  feedback: ListenFeedBack;
+};
+
+export default class ListenCard extends React.Component<
+  ListenCardProps,
+  ListenCardState
+> {
+  APIService: APIService;
   playListen: (listen: Listen) => void;
 
   constructor(props: ListenCardProps) {
     super(props);
 
+    this.state = {
+      feedback: 0,
+    };
+
+    this.APIService = new APIService(
+      props.apiUrl || `${window.location.origin}/1`
+    );
+
     this.playListen = props.playListen.bind(this, props.listen);
   }
 
+  handleClickLove = async () => {
```

Comment:
This and handleClickHate should be merged into one function with feedback as arg